OpenSim/Common/Exception.h

```diff
@@ -67,14 +110,31 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
```

Comment:
Shrinidhi, this "custom message separators" idea is a design feature that we haven't discussed and don't need currently. Please remove it from the Exception class here in the header; customization of formatting should be discussed in the new issue you are going to create for a more comprehensive revamping of OpenSim's error handling system.

> I did not want to change the access specifiers for the existing members in Exception class. 

Please go ahead and do so. Whenever you make a change to old code it is good to correct any small problems you find. Hopefully that way the code gets better over time rather than rotting away. In any case now that you have introduced nicer formatting for the (file,line,func) stuff no one should be accessing those data members except the Exception class itself. If someone is doing something weird, you'll find out when you change protected to private here and then we can correct it.